### PR TITLE
Add sg.Text as available field element / Add edit-protecting a button

### PR DESCRIPTION
**Pull request by ssweber**
Only needed 1 change. 

For update_element_states:
Didn't add to update_event_states, because you can't disable text. 
Didn't know about adding it to visible=visible section.
I updated the update_event_states to be type(element) in to match.

### CODE REVIEW

File: examples/journal_internal.py [hunk 1]
```diff
@@ -72,6 +76,11 @@
 # Requery the data since we made changes to the sort order
 frm['Journal'].requery()
 
+# To edit-protect a Calendar button (or any button), call map_elment()
+frm.map_element(win['datepicker'], frm['Journal'], column=None)
```
Comment (ssweber) [inline examples/journal_internal.py:80]: Or we could add a separate function map_edit_protect(sg.Element, Dataset)?

**Comment by ssweber** [comment]
Sorry, I had to add the Edit-protect a sg.CalendarButton to the other fix, because it had merge-conflicts. If you don't like the sg.CalendarButton, I can repull just the sg.Text one.

**Comment by ssweber** [comment]
Don't merge this yet, somehow it screws up the Quick Editor... looking into it

**Comment by ssweber** [comment]
k, its working now, and I made sure wouldn't have issues elsewhere

**Comment by PySimpleSQL** [comment]
I actually think it's better off done on the users end anyway.  Really all
that needs done is to set the elements state to not frm.edit_protect in the
main loop of the users application to edit protect anything.  Perhaps an
example would be more appropriate.  Or a helper function in pysimplesql:

def edit_protect_element(self):
    return not self.edit_protect


Then the user can just use it in an element update on the main loop to
enable)disable the element

**Pull request closed without merging**